### PR TITLE
Laravel 10 compatibility

### PR DESCRIPTION
**Pull request by SimonBarrettACT**
Packages have been updated to be compatible with Laravel 10.

### CODE REVIEW

**Comment by SimonJnsson** [comment]
Hey @SimonBarrettACT, thanks for contributing!

Could you please also add Laravel 10 to the github [workflow tests](https://github.com/Morning-Train/data-transfer-object-casters/blob/c6ac3fe538785c8e55516794595b3b766aefc8fd/.github/workflows/run-tests.yml#LL17C10-L17C10), so we can ensure the checks are passing for Laravel 10 aswell?

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -14,13 +14,15 @@ jobs:
...
-        laravel: [8.*, 9.*]
+        laravel: [8.*, 9.*, 10.*]
         stability: [prefer-lowest, prefer-stable]
         include:
           - laravel: 8.*
             testbench: ^6.6
           - laravel: 9.*
             testbench: 7.*
+          - laravel: 10.*
+            testbench: 8.*
```
Comment (SimonJnsson) [inline .github/workflows/run-tests.yml:31]: We should exclude certain combinations of php and laravel. Something like this.

```suggestion
        exclude:
          - laravel: 8.*
            php: 8.2
          - laravel: 10.*
            php: 8.0
```

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -14,13 +14,15 @@ jobs:
       matrix:
         os: [ubuntu-latest, windows-latest]
         php: [8.0]
```
Comment (SimonJnsson) [inline .github/workflows/run-tests.yml]: Let's make sure we run the checks for 3 latest php versions,

```suggestion
        php: [8.0, 8.1, 8.2]
```

Comment (SimonBarrettACT) [reply .github/workflows/run-tests.yml]: Sorry for the delay, I have now incorporated your suggestions.

**Review by SimonJnsson — APPROVED** [review]
Thanks for contributing!

**Pull request merged**